proposals/extensions.md

```diff
@@ -126,6 +126,8 @@ public static class Enumerable
 }
 ```
 
+It is an error to specify the `partial` modifier on a member of an extension declaration.
```

Comment:
It would be good to add some explanation for why we are restricting this.

Reply:
I can see some interesting source generator use cases arising from partial members in extensions.

Reply:
We're trimming down on complexity. We'll add if we can identify some useful scenarios. Please drop a note if you know of some. For example, do some folks already use partial extension methods?